windows/src/engine/keyman32/K32_DBG.CPP

```diff
@@ -304,4 +304,12 @@ char *Debug_UnicodeString(PWSTR s, int x) {
   }
   //WideCharToMultiByte(CP_ACP, 0, buf, -1, bufout, 128, NULL, NULL);
   return bufout[x];
-}
\ No newline at end of file
+}
+
+#ifdef _DEBUG
+void _OutputThreadDebugString(char *s) {
+  char buf[256];
+  wsprintf(buf, "[%d]: %s\n", GetCurrentThreadId(), s);
+  OutputDebugString(buf);
```

Comment:
It's not too dangerous since we do control what debug strings are possible within our code, but it might be wise to add a string-length bounds check anyway.